# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/dom/html/HTMLCanvasElement.h	Mon Jun 08 20:34:15 2020 +0000
@@ -29,7 +29,6 @@
 class ClientWebGLContext;
 
 namespace layers {
-class AsyncCanvasRenderer;
 class CanvasRenderer;
 class CanvasLayer;
 class Image;
@@ -123,7 +122,6 @@
                                 public SupportsWeakPtr<HTMLCanvasElement> {
   enum { DEFAULT_CANVAS_WIDTH = 300, DEFAULT_CANVAS_HEIGHT = 150 };
 
-  typedef layers::AsyncCanvasRenderer AsyncCanvasRenderer;
   typedef layers::CanvasRenderer CanvasRenderer;
   typedef layers::CanvasLayer CanvasLayer;
   typedef layers::Layer Layer;
@@ -332,23 +330,14 @@
   layers::LayersBackend GetCompositorBackendType() const;
 
   void OnVisibilityChange();
-
   void OnMemoryPressure();
-
   void OnDeviceReset();
 
-  static void SetAttrFromAsyncCanvasRenderer(AsyncCanvasRenderer* aRenderer);
-  static void InvalidateFromAsyncCanvasRenderer(AsyncCanvasRenderer* aRenderer);
-
   already_AddRefed<layers::SharedSurfaceTextureClient> GetVRFrame();
   void ClearVRFrame();
 
   bool MaybeModified() const { return mMaybeModified; };
 
-  AsyncCanvasRenderer* GetAsyncCanvasRenderer();
-
-  layers::OOPCanvasRenderer* GetOOPCanvasRenderer();
-
  protected:
   virtual ~HTMLCanvasElement();
 
@@ -392,8 +381,7 @@
   RefPtr<HTMLCanvasPrintState> mPrintState;
   nsTArray<WeakPtr<FrameCaptureListener>> mRequestedFrameListeners;
   RefPtr<RequestedFrameRefreshObserver> mRequestedFrameRefreshObserver;
-  RefPtr<AsyncCanvasRenderer> mAsyncCanvasRenderer;
-  RefPtr<layers::OOPCanvasRenderer> mOOPCanvasRenderer;
+  RefPtr<CanvasRenderer> mCanvasRenderer;
   RefPtr<OffscreenCanvas> mOffscreenCanvas;
   RefPtr<HTMLCanvasElementObserver> mContextObserver;
 